tests/checkers/unittest_non_ascii_name.py

```diff
@@ -0,0 +1,312 @@
+import sys
+from typing import Iterable, Optional
+
+import astroid
+import pytest
+from astroid import nodes
+
+import pylint.checkers.non_ascii_names
+import pylint.interfaces
+import pylint.testutils
+
+
+class TestNonAsciiChecker(pylint.testutils.CheckerTestCase):
+    CHECKER_CLASS = pylint.checkers.non_ascii_names.NonAsciiNamesChecker
+    checker: pylint.checkers.non_ascii_names.NonAsciiNamesChecker
+
+    @pytest.mark.skipif(
+        sys.version_info < (3, 8), reason="requires python3.8 or higher"
+    )
+    def test_kwargs_and_position_only(self):
```

Comment:
Isn't this a duplicate of the the functional test in which we also test the location parameters of the message? If we have access to all four I don't think we also need a test whether we have the correct `node` (we don't for 99% of the other messages).

Reply:
For people like me that aren't fluent in AST this helps a lot to see what is going on.
Also in opposit to the functional test I define my expectations first and see if they match.

Do you really look in every functional test, compare the line endings and colums and see if they make sense?
I doubt that.
Especially as they generated automatically.

For me the functional test fell much more like Regression and Integration Tests, that IMO still do not replace Unit test.

But honestly -> are we really discussing now that I add **too many** tests 😉  ?

Reply:
> Do you really look in every functional test, compare the line endings and columns and see if they make sense? I doubt that.

This is something we have to do only once. When we update the functional output we check that the line/column is correct, for a single node, maybe multiple node if there was a particular issue opened about line/column values. The column is very seldom wrong because the information come from the AST and if the ast is wrong, well, we can't do much about it anyway. And if one node is correct the chance that all node are correct is very very very high.

Defining the expected line / column in a ``CheckerTestCase`` is very tedious, at least as tedious as checking the expected line / column in the functional test. But we can upgrade the expected output automatically with functional test once we know that the value are correct. Also reading a functional test is easier to understand by far. By the way ``CheckerTestCase`` is not a unit test, it's a... checker test. Basically the same thing than functional test but with 15 lines of plumbing for each expected message and only one checker instead of multiple (and you can disable other checker's message in functional if you want). Imo the only valid use if if you use parametrize it's the only way to be denser than functional tests.

> But honestly -> are we really discussing now that I add too many tests 😉 ?

There's such a thing as too many tests: if the tests are duplicated or test the same code construct over and over, then we might as well remove the duplicate, save time in our test suite, and save time if we're going to need to update the test later.

Reply:
As I see it:
There are function tests, that test the complete pylint architecture works as expected, i.e. configuration settings messages with the correct line, col, etc....

Each of these tests takes approx 45ms on my machine. They are hard to use for non regular pylint developers like me, as helping Tools like shell completion and "Click to run on test" within an IDE like PyCharm don't work (or other useful features like going to the executed code if the test fails).

On the other hand there are the normal pytest `CheckerTestCases`, all helpers (i.e. code completion in the shell, IDE functions) work as expected. 
Using them, not all things of pylint (settings, lines, etc) work correctly/are considered. Which means they have little side effects and run fast a blink (all unittest together for my checker take approx 1ms on my Machine, a fraction of just a single function test!).
So I am testing only a method, that is kind of mocked and not a complete run -> feels damn much like a unittest to me.

For me as a contributor not fluent in the internals of neither pylint nor (and espcially) of AST/astroid they help a big deal understanding what is going on, without setting a number of brakepoints everywhere:
Looking at the unittests I know exactly which methods I would need to adopt and how the parameters are extracted.

So of course I can remove them (besides the import unit tests), as "duplicated" functional tests exist.

But I hope you (@Pierre-Sassoulas and @DanielNoord ) understand that from my view as a contributor it just feels wrong to remove these tests.

Still I removed two tests that were really just duplicates (import star and class). 

But I really would like keep the remainig checks as I they are very helpful for contributor IMO.

Reply:
Thank for bringing the speed issue to my attention. Maybe related to https://github.com/PyCQA/astroid/issues/1320. Anyway, I'll resolve because arguing about this is longer than the time it'll take to move to functional if we need to modify it later.